src/main/host/process.c

```diff
@@ -776,35 +776,30 @@ static void _process_handleTimerResult(Process* proc, gdouble elapsedTimeSec) {
 static gint _process_getArguments(Process* proc, gchar** argvOut[]) {
     gchar* threadBuffer;
 
-    GQueue *arguments = g_queue_new();
+    gchar* arguments = NULL;
 
     /* first argument is the name of the program */
     const gchar* pluginName = _process_getPluginName(proc);
-    g_queue_push_tail(arguments, g_strdup(pluginName));
+    arguments = g_strdup(pluginName);
 
     /* parse the full argument string into separate strings */
     if(proc->arguments && proc->arguments->len > 0 && g_ascii_strncasecmp(proc->arguments->str, "\0", (gsize) 1) != 0) {
-        gchar* argumentString = g_strdup(proc->arguments->str);
-        gchar* token = strtok_r(argumentString, " ", &threadBuffer);
-        while(token != NULL) {
-            gchar* argument = g_strdup((const gchar*) token);
-            g_queue_push_tail(arguments, argument);
-            token = strtok_r(NULL, " ", &threadBuffer);
-        }
-        g_free(argumentString);
+        arguments = g_strconcat(arguments, " ", proc->arguments->str, NULL);
     }
```

Comment:
I think this is leaking memory. The pointer returned by `g_strdup` is being overwritten after the `g_strconcat` call and is never `free()`d. How about this instead:

```c
    gchar* arguments = NULL;
    if(proc->arguments && proc->arguments->len > 0 && g_ascii_strncasecmp(proc->arguments->str, "\0", (gsize) 1) != 0) {
        arguments = g_strconcat(pluginName, " ", proc->arguments->str, NULL);
    } else {
        arguments = g_strdup(pluginName);
    }
    ....
    g_free(arguments);
```

Reply:
thanks for pointing that out. done!